### ISSUE TITLE
[V5]The UIEffect Reverse property does not work

### ISSUE THREAD

**Issue by xfvvvv** [mob-sakai/UIEffect#283]
Hello,

The Reverse property does not work in UIEffect. It is not possible to play the effect in reverse. For example, when using the Dissolve effect, you can only play it in the forward direction.

Thank you!

**Comment by mob-sakai** [comment]
Thank you for your reporting!

**Comment by mob-sakai** [comment]
It is scheduled to be released in version 5.1.0! :+1:

**Comment by alexxtaurus83** [comment]
Any updates? still not working. Tried to check develop-exp - not working too.
Tring to play Dissolve effect,

var tweener = gameObject.AddComponent<UIEffectTweener>();                    
tweener.wrapMode = UIEffectTweener.WrapMode.Once;
tweener.duration = 5f;
tweener.playOnEnable = PlayOnEnable.Reverse;
tweener.direction = Direction.Reverse;
tweener.PlayReverse();

**Comment by mob-sakai** [comment]
There are two approaches to implement the intended behavior:

### Use UIEffectTweener.SetTime

```cs
var tweener = gameObject.AddComponent<UIEffectTweener>();
tweener.wrapMode = UIEffectTweener.WrapMode.Once;
tweener.duration = 5f;
tweener.playOnEnable = PlayOnEnable.Reverse;
tweener.direction = Direction.Reverse;
tweener.SetTime(5); // Added
tweener.PlayReverse();
```

### Use UIEffectTweener.curve

```cs
var tweener = gameObject.AddComponent<UIEffectTweener>();
tweener.wrapMode = UIEffectTweener.WrapMode.Once;
tweener.duration = 5f;
tweener.curve = AnimationCurve.Linear(0,1,1,0); // Added
tweener.PlayForward();
```

**Comment by alexxtaurus83** [comment]
Worked this way:
 var tweener = wikiItemPresenter.gameObject.AddComponent<UIEffectTweener>();
 tweener.wrapMode = UIEffectTweener.WrapMode.Once;
 tweener.duration = effectDuration;
 tweener.curve = AnimationCurve.Linear(0, 1, 1, 0); // Added
 tweener.PlayForward();

Can I use stable version 5.0.2 to avoid develop-exp?

**Comment by mob-sakai** [comment]
👍 

> Can I use stable version 5.0.2 to avoid develop-exp?

Hmm, I can't guarantee that code will work with UIEffect v5.0.2, but feel free to try it.  
By the way, `develop-exp` is expected to be released as 5.1.0 soon.

**Comment by alexxtaurus83** [comment]
Thank you for your work!
